File: Core/Entities/DeliveryMethod.cs
namespace Core.Entities;

/// <summary>
/// Represents a delivery method available for orders.
/// </summary>
public class DeliveryMethod : BaseEntity
{
  /// <summary>
  /// Short name of the delivery method.
  /// </summary>
  public required string ShortName { get; set; }

  /// <summary>
  /// Estimated delivery time for the method.
  /// </summary>
  public required string DeliveryTime { get; set; }

  /// <summary>
  /// Detailed description of the delivery method.
  /// </summary>
  public required string Description { get; set; }

  /// <summary>
  /// Price of the delivery method.
  /// </summary>
  public decimal Price { get; set; }
}

File: Client/src/app/shared/models/deliveryMethod.ts
/**
 * Represents a delivery method with details such as time and price.
*/
export type DeliveryMethod = {
  /** A short name to identify the delivery method. */
  shortName: string;

  /** The expected time for delivery. */
  deliveryTime: string;

  /** A description of the delivery method. */
  description: string;

  /** The price of the delivery method. */
  price: number;

  /** The unique identifier for the delivery method. */
  id: number;
}